doc/source/user_guide.md

```diff
@@ -0,0 +1,318 @@
+# User guide
+                            
+
+## How to reset your password
+
+
+**NB**: This section is only available if the use of emails to login is
+activated in the global configuration. It is not the case by default.
+
+If you forgot your password or if you need to set one, first call the
+forgotten_password endpoint:
+
+<!-- It is python, just doing this so that test-integration does not run this code (need mail config to run)  -->
+
+```javascript
+from avatars.client import ApiClient
+
+client = ApiClient(base_url=os.environ.get("BASE_URL"))
+client.forgotten_password("yourmail@mail.com")
+```
+You’ll then receive an email containing a token. This token is only
+valid once, and expires after 24 hours. Use it to reset your password:
+
+```javascript
+from avatars.client import ApiClient
+
+client = ApiClient(base_url=os.environ.get("BASE_URL"))
+client.reset_password("yourmail@mail.com", "new_password", "new_password", "token-received-by-mail")
+```
+
+You’ll receive an email confirming your password was reset.
+
+## How to log in to the doc/source/user_guide.md server 
+
+```python
+import os
+
+# This is the client that you'll be using for all of your requests
+from avatars.client import ApiClient
+
+# The following are not necessary to run avatar but are used in this tutorial
+import pandas as pd
+import io
+
+# Change this to your actual server endpoint, e.g. base_url="https://avatar.company.com"
+client = ApiClient(base_url=os.environ.get("BASE_URL"))
+client.authenticate(
+    username="username", password=os.environ.get("AVATAR_PASSWORD", "strong_password")
+)
+```
+
+## How to upload a dataframe
```

Comment:
```suggestion
## How to upload a data
```
You are wanting to expose how data is uploaded, both as a dataframe, and as a csv. So the title has to be generic